pkg/controller/masteruserrecord/sync.go

```diff
@@ -185,7 +186,7 @@ func (s *Synchronizer) alignReadiness() (bool, error) {
 	if condition.IsNotTrue(s.record.Status.Conditions, toolchainv1alpha1.MasterUserRecordUserProvisionedNotificationCreated) {
 		notification := &toolchainv1alpha1.Notification{
 			ObjectMeta: v1.ObjectMeta{
-				Name:      s.record.Name + "-provisioned",
+				Name:      s.record.Name + "-provisioned" + "-" + strconv.FormatInt(v1.Now().UnixNano(), 10),
```

Comment:
I think it would be preferable here to use `fmt.Sprintf()` here instead of a multi-string concatenation.

Reply:
This is removed in favour of using GenerateName